src/ORM/Query.php

```diff
@@ -31,6 +33,41 @@
  * into a specific iterator that will be responsible for hydrating results if
  * required.
  *
+ * @see CollectionInterface For a full description of the collection methods supported by this class
+ * @method CollectionInterface each(callable $c) Passes each of the query results to the callable
+ * @method CollectionInterface filter(callable $c = null) Keeps the results using passing the callable test
+ * @method CollectionInterface reject(callable $c) Removes the results passing the callable test
+ * @method bool every(callable $c) Returns true if all the results pass the callable test
+ * @method bool some(callable $c) Returns true if at least one of the results pass the callable test
+ * @method CollectionInterface map(callable $c) Modifies each of the results using the callable
+ * @method mixed reduce(callable $c, $zero = null) Folds all the results into a single value using the callable.
+ * @method CollectionInterface extract($field) Extracts a single column from each row
+ * @method mixed max($field, $type = SORT_NUMERIC) Returns the maximum value for a single column in all the results.
+ * @method mixed min($field, $type = SORT_NUMERIC) Returns the minimum value for a single column in all the results.
+ * @method CollectionInterface groupBy(string|callable $field) In-memory group all results by the value of a column.
```

Comment:
The multiple type syntax (`string|callable`) will cause problems in at least PhpStorm, it will be marked as an error. While this is clearly a bug in the IDE, I'm wondering whether such IDE compatibilities should be considered?